docs/source/upgrade/index.rst

```diff
@@ -82,13 +83,109 @@ has one *nonconsuming* choice that accepts a *Coin* contract, archives
 this *Coin* contract and creates a *CoinWithAmount* contract with
 *amount* set to 1.
 
-.. literalinclude:: daml/Upgrade.daml
+.. literalinclude:: example/coinV2/daml/UpgradeFromCoinV1.daml
   :language: daml
   :start-after: -- UPGRADE_AGREEMENT_BEGIN
   :end-before: -- UPGRADE_AGREEMENT_END
 
-.. TODO Turn this into a full example project with multiple packages
-.. and explain builds and deployment.
 
-.. Building and deploying
-.. ======================
+Building and deploying version-1
+================================
+
+Let's see everything in action by following the steps to build and deploy version-1 of our coin. After this we'll see how to deploy and upgrade to version-2.
+
+First we'll need a sandbox ledger to which we can deploy.
+
+.. code-block:: none
+
+   $ daml sandbox --port 7001
+
+The version-1 project is named ``coinV1``. It contains just the original *Coin* definition template ``Coin``. Here is the ``daml.yaml`` setup
+
+.. literalinclude:: example/coinV1/daml.yaml
+  :language: yaml
+  :start-after: # BEGIN
+  :end-before: # END
+
+To make the example complete, we need a way to create some coins which have different parties for the issuer and owner. We take the standard approach of adding a :ref:`Propose-Accept workflow <intro propose accept>` to our example.
```

Comment:
The text here does not quite match the code. I would have expected something like a `CoinProposal` signed by the issuer and then the `owner` can accept it. Instead you have an extra step: First, the issuer has to create a self-issues `Coin`, then they have to create a `TransferProposal` which then has to be accepted by the owner.

Given that this is not the focus of this tutorial I would keep it as simple as possible and not include transfers. The fact that the upgraded coin is no longer transferable is also a bit confusing and while that’s fixable it only adds more complexity.

Reply:
I agree.
I don't really want to present the Transfer. But I do want to use it in the example steps.
So it will have to exist in the example